### PR TITLE
Helmfile ibft2

### PR DESCRIPTION
**Pull request by joshuafernandes**

### CODE REVIEW

File: helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  labels:
+    app: {{ template "besu-monitoring.name" . }}
+    chart: {{ template "besu-monitoring.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    namespace: monitoring
+    component: prometheus
+  namespace: monitoring
+spec:
+  type: NodePort
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml:14]: Same question here re: NodePort vs ClusterIP + Ingress

File: helmfile/ibft2/charts/besu-monitoring/templates/grafana-service.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    component: grafana
+  namespace: monitoring
+spec:
+  type: NodePort
+  selector:
+    component: grafana
+  ports:
+    - port: 3000
+      targetPort: 3000
+      nodePort: 30030
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-monitoring/templates/grafana-service.yaml:20]: Should this be configurable via values?

File: helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  selector:
+    component: prometheus
+  ports:
+    - port: 9090
+      targetPort: 9090
+      nodePort: 30090
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml:20]: Should this be configurable via values?

File: helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml [hunk 4]
```diff
@@ -2,29 +2,30 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ include "besu-node.fullname" . }}-config
+  name: {{ include "besu-node.fullname" . }}-config-toml
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml:5]: Not a comment on this line specifically, but the entire file: The complexity of breaking this out into values makes me wonder if we shouldn't just use the include file trick that you used above for the `prometheus.yml` file

File: helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml [hunk 5]
```diff
@@ -33,87 +34,81 @@ data:
...
-    miner-enabled={{ .Values.node.miner.enabled }}
-    miner-coinbase={{ .Values.node.miner.coinbase | quote }}
-    miner-extra-data={{ .Values.node.miner.extra_data | quote }}
-    {{- end -}}
+    max-peers={{ .Values.node.p2p.maxPeers }}
+    {{ end }}
 
     {{ if or .Values.node.rpc.enabled .Values.node.graphql.enabled .Values.node.ws.enabled }}
-    host-whitelist={{ .Values.node.http.whitlist }}
+    host-whitelist={{ .Values.node.http.whitelist }}
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml:53]: Nice catch

File: helmfile/ibft2/charts/besu-node/templates/node-statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+            protocol: UDP
+          - name: metrics
+            containerPort: {{ .Values.node.metrics.port }}
+            protocol: TCP
+        command:
+          - /bin/sh
+          - -c
+        args:
+          - |
+            exec /opt/besu/bin/besu \
```
Comment (benjamincburns) [inline helmfile/ibft2/charts/besu-node/templates/node-statefulset.yaml:151]: How would I inject JVM runtime args (e.g. -Xmx, JFR options, etc?).

Perhaps we need a value to assign to `BESU_OPTS` or whatever that env var name is?

Comment (benjamincburns) [reply helmfile/ibft2/charts/besu-node/templates/node-statefulset.yaml:151]: Similarly, it'd be nice if I could add other args here via values for flags like `--Xsecp256k1-native-enabled=true`, etc.

File: helmfile/ibft2/helmfile.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  - ./helmfile.d/00-genesis.yaml
+
+
+# The desired states of Helm releases.
+releases:
+  - name: monitoring
+    namespace: {{ .Namespace }}
+    chart: ./charts/besu-monitoring
+
+  - name: bootnode-1
```
Comment (benjamincburns) [inline helmfile/ibft2/helmfile.yaml:35]: Do we still need this + bootnode 2 if the validators are also serving as bootnodes?

File: helmfile/ibft2/charts/besu-genesis/templates/genesis-job.yaml [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          volumeMounts:
+            - name: raw-config
+              mountPath: /raw-config
+            - name: generated-config
+              mountPath: /generated-config
+          command:
+            - /bin/bash
+            - -c
+          args:
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-genesis/templates/genesis-job.yaml:85]: Yep, this is k8s specific only - here'ls the only place we create keys, geneis etc. All the others are pre created and put into values.yaml

File: helmfile/ibft2/charts/besu-monitoring/templates/_helpers.tpl [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "besu-monitoring.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-monitoring/templates/_helpers.tpl:11]: yep, good point. I'll add this in now

File: helmfile/ibft2/charts/besu-monitoring/templates/grafana-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+            - name: POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: GF_SECURITY_ADMIN_USER
+              value: "admin"
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-monitoring/templates/grafana-deployment.yaml:47]: We've got notes in the Readme for this to say configure to suit your needs via env vars - I've updated the main readme with a link to show the setups grafana provides.

File: helmfile/ibft2/charts/besu-monitoring/prometheus.yml [hunk 11]
```diff
@@ -0,0 +1,61 @@
+---
+global:
+  scrape_interval:     15s
+  evaluation_interval: 15s
+alerting:
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-monitoring/prometheus.yml:5]: Could do, problem is where we send the default alert to - email requires a token + email, same for slack etc.

File: helmfile/ibft2/charts/besu-node/templates/node-statefulset.yaml (hunk 6 above)
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-node/templates/node-statefulset.yaml:151]: BESU_OPTS it is as an env var would do it :) we can add one with some sensible defaults as an example ?

File: helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml (hunk 5 above)
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-node/templates/config-toml-configmap.yaml:53]: :)

File: helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml (hunk 3 above)
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-monitoring/templates/prometheus-service.yaml:20]: Yep, we can do this one is a seperate PR for the lot

File: helmfile/ibft2/helmfile.yaml (hunk 7 above)
Comment (joshuafernandes) [reply helmfile/ibft2/helmfile.yaml:35]: Kind of - I added this as an expection to our normal deployments to show how to use separate bootnodes. There is a note about minikube in the readme

File: helmfile/ibft2/charts/besu-monitoring/grafana-dashboard-besu.json [hunk 12]
```diff
@@ -0,0 +1,884 @@
+{
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helmfile/ibft2/charts/besu-monitoring/grafana-dashboard-besu.json:1]: yep,its the same one that lives in every single chart - balancing act of making it easy to use for people (with everything contained in one folder they can customise) and a step towards prod. Maybe in the next monitoring pr we extract this to a single place?

**Pull request merged**